src/github/actions-provider.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Manage the versions used for GitHub Actions used in steps
+ */
+export class GitHubActionsProvider {
+  private actions = new Map();
```

Comment:
I prefer the map types to be fully specified, as I'm not left to ponder what the key & value is inferred to be from reading the rest of the code.

```suggestion
  private actions = new Map<string, string>();
```